src/bindings/jna/plugins/sorted/src/main/java/org/libelektra/plugin/SortedPlugin.java

```diff
@@ -0,0 +1,157 @@
+package org.libelektra.plugin;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.libelektra.*;
+
+public class SortedPlugin implements Plugin {
+
+  private static final String PLUGIN_NAME = "Sorted";
+  private static final String META_SORTED = "check/sorted";
+  private static final String META_SORTED_DIRECTION = "check/sorted/direction";
+
+  private enum Direction {
+    ASC,
+    DESC
+  }
+
+  @Nonnull
+  @Override
+  public String getName() {
+    return PLUGIN_NAME;
+  }
+
+  @Override
+  public int open(KeySet config, Key errorKey) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int get(KeySet keySet, Key parentKey) {
+    if (isPluginMetadataRequested(parentKey)) {
+      SortedMetadata.appendAllTo(keySet);
+    }
+
+    checkForSortingErrors(keySet, parentKey, parentKey::addWarning);
+
+    return STATUS_SUCCESS;
+  }
+
+  private boolean isPluginMetadataRequested(Key parentKey) {
+    return parentKey.isBelowOrSame(Key.create(PROCESS_CONTRACT_ROOT));
+  }
+
+  @Override
+  public int set(KeySet keySet, Key parentKey) throws KDBException {
+    return checkForSortingErrors(keySet, parentKey, parentKey::setError)
+        ? STATUS_ERROR
+        : STATUS_SUCCESS;
+  }
+
+  private boolean checkForSortingErrors(
+      KeySet keySet, Key parentKey, BiFunction<ErrorCode, String, Key> addErrorFunction) {
+    AtomicBoolean foundError = new AtomicBoolean(false);
```

Comment:
I used it, because IntelliJ suggested this change. The reason behind it is the stream which is used in this method. If this stream runs in parallel, there may be issues though I think one needs to specify this explicitly.

Reply:
That is a weird suggestion from IntelliJ. I get the idea, but there is no `Stream` in this function. The `forEach` you used comes directly from `Iterable` and according to [the docs](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)) it is basically the same as a for-each loop.

But you can leave it the way it is. It shouldn't really matter much anyway

Reply:
Ah, yes of course. Ok, then i will leave it that way, maybe you want to use it in parallel in the future. ;)